api/modules/dataSyncs.js

```diff
@@ -12,6 +13,44 @@ const { GITHUB, TOGGL } = require('../config/credentials')
 
 const dataSyncs = module.exports = (() => {
 
+    const importInvoicelyCsvToStripe = async () => {
+        const invoiceFile = INVOICELY_CSV_PATH
+        const csvFile =
+            await amazon.fetchFile({ file: invoiceFile })
+                .then(file => {
+                    return invoicelyCodebase.modelCSV(file)
+                })
+                .catch(err => {
+                    console.log('error', err);
+                    return err.message
+                })
+        const customerObject = await stripe.listAllCustomers()
+        const customersFromCsv = []
+        try {
+            csvFile.map(async csvData => {
+                let customerInformation
+                const customer = customerObject.data.find(customerData => customerData.name == csvData.Client)
+                customerInformation = customer
+                csvData.Total = csvData.Total.replace(/,/g, '')
+                if (customer) {
+                    customerInformation = customer
+                } else if (!customersFromCsv.includes(csvData.Client)) {
+                    customerInformation = await stripe.createCustomer({ name: csvData.Client, email: null })
+                    customersFromCsv.push(csvData.Client)
+                }
+                await stripe.createInvoice({
+                    amount: parseInt(csvData.Total),
+                    external_uuid: customerInformation.id,
+                    currency: csvData.Currency
+                })
```

Comment:
The only way to set a date is using the due date and if we use this parameter it will create the invoice as a pending one, so when the date is already the invoice is charged, the invoices with an older date Stripe just ignore them.